upload-api/test/helpers/ipni-service.js

```diff
@@ -0,0 +1,45 @@
+import { base58btc } from 'multiformats/bases/base58'
+import { error, ok } from '@ucanto/server'
+import { useIPNIService, MultihashesQueue, BlocksCarsPositionStore } from '../../external-services/ipni-service.js'
+import { blocksCarsPositionTableProps } from '../../tables/index.js'
+import { createTable, createQueue } from '../helpers/resources.js'
+import { collectQueueMessages } from './queue.js'
+import { RecordNotFound } from '@web3-storage/upload-api/errors'
+
+/**
+ * @param {{ sqs: import('@aws-sdk/client-sqs').SQSClient, dynamo: import('@aws-sdk/client-dynamodb').DynamoDBClient }} config
+ */
+export const createTestIPNIService = async ({ sqs, dynamo }) => {
+  const queueURL = await createQueue(sqs, 'multihashes')
+  const multihashesQueue = new MultihashesQueue({
+    client: sqs,
+    url: queueURL
+  })
+
+  const tableName = await createTable(dynamo, blocksCarsPositionTableProps)
+  const carsBlocksPositionStore = new BlocksCarsPositionStore({
+    client: dynamo,
+    name: tableName
+  })
+
+  const messages = new Set()
+  return Object.assign(
+    useIPNIService(multihashesQueue, carsBlocksPositionStore),
+    {
+      /** @param {import('multiformats').MultihashDigest} digest */
+      async query (digest) {
```

Comment:
Nit: I think calling this `has` or `contains` would be far less confusing

Reply:
Yes I did consider this. However I don't believe there's an explicit IPNI API for that, and I thought at some point we might want query to actually return the result of querying IPNI for a multihash.

Should have changed in w3up types before now...